screed/tests/test_attriberror.py

```diff
@@ -0,0 +1,85 @@
+import screed
+from screed.DBConstants import fileExtension
+import os
+
+
+class nostring:
+    def __init__(self):
+        self.exists = True
+
+    def __str__(self):
+        raise AttributeError
```

Comment:
`__str__`, `__repr__`, `__cmp__` are not attributes but special methods. It is inappropriate to throw `AttributeError` from them. I think you wrote this to test https://github.com/ged-lab/screed/blob/fix/896/screed/screedRecord.py#L100; that code shouldn't be catching an AttributeError as that shouldn't occur from calling `str(object)`: https://docs.python.org/2/library/functions.html#str

According to http://ci.ged.msu.edu/job/screed/48/cobertura/_default_/screed_screedRecord_py/ all uses of catching AttributeError's are untested. The original commit is f42562812fde9be6ee410121170df215d2e963df

So we need to determine if that approach is correct. I suspect it is not. Perhaps @ctb or @acr have an opinion